src/api/java/com/minecolonies/api/colony/fields/AbstractFieldMatcher.java

```diff
@@ -0,0 +1,97 @@
+package com.minecolonies.api.colony.fields;
+
+import com.minecolonies.api.colony.fields.registry.FieldRegistries;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.FriendlyByteBuf;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Abstract field matcher implementation, can be overridden without having to do
+ * anything else to get a basic "position only" field matcher.
+ */
+public abstract class AbstractFieldMatcher implements IFieldMatcher
+{
+    /**
+     * The field type.
+     */
+    private final @NotNull FieldRegistries.FieldEntry fieldType;
+
+    /**
+     * The position of the field.
+     */
+    private final @NotNull BlockPos position;
+
+    /**
+     * Default constructor.
+     *
+     * @param fieldType the field type.
+     * @param position  the position of the field.
+     */
+    protected AbstractFieldMatcher(@NotNull FieldRegistries.FieldEntry fieldType, @NotNull BlockPos position)
+    {
+        this.fieldType = fieldType;
+        this.position = position;
+    }
+
+    @Override
+    public final @NotNull FieldRegistries.FieldEntry getFieldType()
+    {
+        return fieldType;
+    }
+
+    @Override
+    public final @NotNull BlockPos getPosition()
+    {
+        return position;
+    }
+
+    @Override
+    public boolean matches(@Nullable final IField other)
```

Comment:
Why is this necessary? Cant we just have that logic in the field itself?

Reply:
The idea of the field matcher was that you could create an instance of them, without needing the data of the entire field, mainly meant for messages back to the server.
For example when I want to change the plant on a farm field, I'd make a farm field matcher and give it the position, instead of having to give it the entire field instance.

Reply:
Fields are so tiny, not worth the added complexity.